UI/basicGUI/src/pages/home/home.ts

```diff
@@ -34,27 +39,93 @@ export class HomePage {
       markerGroup.addLayer(this.marker);
       this.map.addLayer(markerGroup);
 
-      }).on('locationerror', (err) => {
-          console.log(err.message);
-      })
+    }).on('locationerror', (err) => {
+      console.log(err.message);
+    })
   }
 
-    showLocation(){
-      this.getLocation()
+  showLocation() {
+    this.getLocation()
 
-      let alertOfLoc = this.alertCtrl.create({
-        title:"YOUR CURRENT LOCATION",
-        subTitle: "Latitude:" + this.currentLatlng.lat+" Longitude: " +this.currentLatlng.lng,
-        buttons:['GOT IT']
-      });
+    let alertOfLoc = this.alertCtrl.create({
+      title: "YOUR CURRENT LOCATION",
+      subTitle: "Latitude:" + this.currentLatlng.lat + " Longitude: " + this.currentLatlng.lng,
+      buttons: ['GOT IT']
+    });
     alertOfLoc.present();
   }
 
   loadmap() {
+    if (this.map) {
+      this.map.remove();
+    }
     this.map = leaflet.map("map").fitWorld();
     leaflet.tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
       attributions: 'www.tphangout.com',
       maxZoom: 18
     }).addTo(this.map);
+    this.getLocation()
+
+    //this.getPolyLine()
+  }
+
```

Comment:
Merged the split_routing branch - any changes to this file should use that version not this one!